numpy/core/tests/test_regression.py

```diff
@@ -39,12 +39,13 @@ def test_mem_empty(self):
     def test_pickle_transposed(self):
         # Ticket #16
         a = np.transpose(np.array([[2, 9], [7, 0], [3, 8]]))
-        f = BytesIO()
-        pickle.dump(a, f)
-        f.seek(0)
-        b = pickle.load(f)
-        f.close()
-        assert_array_equal(a, b)
+        for proto in range(2, pickle.HIGHEST_PROTOCOL + 1):
+            f = BytesIO()
```

Comment:
I think most of these types of tests should be changed to use `BytesIO` as a context manager, but that is out of scope for this PR.